src/ArrayView.hpp

```diff
@@ -545,18 +602,39 @@ class ArrayView
    * @brief Set all entries in the array to @p value.
    * @tparam POLICY The RAJA policy to use.
    * @param value The value to set entries to.
+   * @note The view is moved to and touched in the appropriate space.
    */
   DISABLE_HD_WARNING
   template< typename POLICY >
   void setValues( T const & value ) const
   {
     auto const view = toView();
-    RAJA::forall< POLICY >( RAJA::TypedRangeSegment< INDEX_TYPE >( 0, size() ), [value, view] LVARRAY_HOST_DEVICE ( INDEX_TYPE const i )
+    RAJA::forall< POLICY >( RAJA::TypedRangeSegment< INDEX_TYPE >( 0, size() ),
+                            [value, view] LVARRAY_HOST_DEVICE ( INDEX_TYPE const i )
       {
         view.data()[ i ] = value;
       } );
   }
 
+  /**
+   * @brief Use memset to set all the values in the array to 0.
+   * @details This is preferred over setValues< POLICY >( 0 ) for numeric types since it is much faster in most cases.
+   *   If the buffer is allocated using Umpire then the Umpire ResouceManager is used, otherwise std::memset is used.
+   * @note The memset occurs in the last space the array was used in and the view is moved and touched in that space.
+   */
+  inline void zero() const
```

Comment:
do we want to allow setting the memory space?

Reply:
Definitely since it can avoid a move. I think the best way to do this is have `registerTouch` allocate in the given space if it didn't already exist and then touch it there. That way all `memcpy` has to do is replace `move( space, true )` with `registerTouch( space )`. I don't really want to add that to this PR though so I'll create an issue.